test/test_pytree.py

```diff
@@ -26,32 +25,15 @@ def __init__(self, x, y):
         self.y = y
 
 
-class TestPytree(TestCase):
-    def test_treespec_equality(self):
-        self.assertTrue(
-            py_pytree.LeafSpec() == py_pytree.LeafSpec(),
-        )
-        self.assertTrue(
-            py_pytree.TreeSpec(list, None, []) == py_pytree.TreeSpec(list, None, []),
-        )
-        self.assertTrue(
-            py_pytree.TreeSpec(list, None, [py_pytree.LeafSpec()])
-            == py_pytree.TreeSpec(list, None, [py_pytree.LeafSpec()]),
-        )
-        self.assertFalse(
-            py_pytree.TreeSpec(tuple, None, []) == py_pytree.TreeSpec(list, None, []),
-        )
-        self.assertTrue(
-            py_pytree.TreeSpec(tuple, None, []) != py_pytree.TreeSpec(list, None, []),
-        )
-
-    def test_flatten_unflatten_leaf(self):
+class TestGenericPytree(TestCase):
```

Comment:
Btw, this PR would be easier to review if we split the testing changes into its own PR